src/main/java/bio/terra/app/controller/SnapshotAccessRequestApiController.java

```diff
@@ -105,6 +107,48 @@ public ResponseEntity<SnapshotAccessRequestResponse> approveSnapshotAccessReques
     return ResponseEntity.ok(snapshotBuilderService.approveRequest(id));
   }
 
+  @Override
+  public ResponseEntity<SnapshotAccessRequestMembersResponse> getSnapshotAccessRequestGroupMembers(
+      UUID id) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    iamService.verifyAuthorization(
+        userRequest, IamResourceType.SNAPSHOT_BUILDER_REQUEST, id.toString(), IamAction.GET);
+    SnapshotAccessRequestResponse request = snapshotBuilderService.getRequest(id);
+    if (request != null && request.getCreatedSnapshotId() == null) {
+      throw new IllegalArgumentException(
+          "Snapshot must be created from this request in order to manage group membership.");
```

Comment:
I think I recommended this but now that I've thought about it more, I think we actually don't need to check if the request is null because the DAO throws a not found error, it doesn't return null.

Reply:
I think this check is necessary. The case this is checking for is one where the request does exist, but it doesn't have a group name set. This will result in `null` being passed to `IamService.addEmailToGroup()`, and I think the exception there is an NPE, which ends up being a 500 error. We need to return a 400, and it would be nice to also return a message telling the user why that request isn't valid for this operation.

I originally suggested doing the check here in the controller, but its intent would be clearer if it was moved into `SnapshotBuilderService`, so it was more obvious why the check is necessary. Something like:
```
  public SnapshotAccessRequestMembersResponse addGroupMember(UUID id, String memberEmail) {
    SnapshotAccessRequestModel model = snapshotRequestDao.getById(id);
    ValidationUtils.requireNotBlank(memberEmail,
        "Snapshot must be created from this request in order to manage group membership.");
```

Reply:
I agree the check for the samGroupName is needed, but not the null check for the request, just like in the sample code you have here.